requirements.txt

```diff
@@ -7,3 +7,4 @@ ply>=3.10
 psutil>=5
 scipy>=1.0
 sympy>=1.3
+python-constraint
```

Comment:
this pass looks good to me, but it's adding an extra dependency for something that is not even invoked during the standard compilation/execution flow. can this be made optional or upon-request or something?

Reply:
I think it makes sense to put in the default pipeline, since its relatively fast. However, we can discuss it later.

Reply:
done in 5faf45c7